Add guide for Helm support (#291)
vmware/terraform-provider-tanzu-mission-control: docs/guides/tanzu-mission-control_helm.md

---
Title: "Helm  Resource"
Description: |-
    Adding Helm resource to a cluster.
---

# Defining Helm

Enable the Helm service when you want to install Helm charts on a cluster or cluster group.

> When you enable the Helm service on a cluster or cluster group, you can then create releases in your cluster from Helm charts stored in the Bitnami repository.

Reviewer: ```suggestion
When you enable the Helm service on a cluster or cluster group, you can then create releases in your cluster from Helm charts stored in the Bitnami Helm repository.
```

Author: Updated PTAL

# Terraform Resources for Helm

## Helm Feature

> The `tanzu-mission-control_helm_feature` resource allows you to enable and disable helm feature to a particular scope through Tanzu Mission Control.

Reviewer: ```suggestion
The `tanzu-mission-control_helm_feature` resource allows you to enable and disable helm feature for a particular scope through Tanzu Mission Control.
```

Author: Updated PTAL

Only after enabling the helm feature User will be able to create the helm release. 

Please refer to [Enable Helm Service on Your Cluster or Cluster Group.][helm-feature]

[helm-feature]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-0927CDC8-A5C1-4FAE-9A7C-8A5D62FDF8D8.html

## Helm Release

The `tanzu-mission-control_helm_release` resource allows you to install, update, get and delete helm chart to a particular scope through Tanzu Mission Control.

Before creating helm charts user needs to enable the helm service on a particular scope cluster or cluster group and for enable user can use `tanzu-mission-control_helm_feature` resource.
The`feature_ref` field of `tanzu-mission-control_helm_release` when specified, ensures clean up of this Terraform resource from the state file by creating a dependency on the Helm feature when the Helm feature is disabled.

> To add a helm charts, you must be associated with the cluster.admin or clustergroup.admin role.

Reviewer: ```suggestion
To create a Helm release, you must be associated with the cluster.admin or clustergroup.admin role.
```

Author: Updated PTAL
